[ERROR] Could not process "/src/phpcms/parser/include/class.pax_phpcms.php" file, due to: "System error: "empty branch"

DEBUG:
 [ERROR] Could not process "/src/phpcms/parser/include/class.pax_phpcms.php" file, due to:
         "System error: "empty branch"

         Stack trace:
         #0 vendor/rector/rector/rules/Php70/EregToPcreTransformer.php(89):
         Rector\Php70\EregToPcreTransformer->_ere2pcre()
         #1 vendor/rector/rector/rules/Php70/EregToPcreTransformer.php(73):
         Rector\Php70\EregToPcreTransformer->ere2pcre()
         #2 vendor/rector/rector/rules/Php70/Rector/FuncCall/EregToPregMatchRector.php(91):
         Rector\Php70\EregToPcreTransformer->transform()
         #3 vendor/rector/rector/rules/Php70/Rector/FuncCall/EregToPregMatchRector.php(172):
         Rector\Php70\Rector\FuncCall\EregToPregMatchRector->processStringPattern()
         #4 vendor/rector/rector/rules/Php70/Rector/FuncCall/EregToPregMatchRector.php(64):
         Rector\Php70\Rector\FuncCall\EregToPregMatchRector->refactorFuncCall()
         #5 vendor/rector/rector/src/Rector/AbstractRector.php(140):
         Rector\Php70\Rector\FuncCall\EregToPregMatchRector->refactor()
         #6 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(114):
         Rector\Rector\AbstractRector->enterNode()
         #7 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(134):
         PhpParser\NodeTraverser->traverseNode()
         #8 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(198):
         PhpParser\NodeTraverser->traverseNode()
         #9 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(105):
         PhpParser\NodeTraverser->traverseArray()
         #10 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(198):
         PhpParser\NodeTraverser->traverseNode()
         #11 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(105):
         PhpParser\NodeTraverser->traverseArray()
         #12 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(198):
         PhpParser\NodeTraverser->traverseNode()
         #13 vendor/rector/rector/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(85):
         PhpParser\NodeTraverser->traverseArray()
         #14 vendor/rector/rector/src/PhpParser/NodeTraverser/RectorNodeTraverser.php(47):
         PhpParser\NodeTraverser->traverse()
         #15 vendor/rector/rector/src/Application/FileProcessor.php(113):
         Rector\PhpParser\NodeTraverser\RectorNodeTraverser->traverse()
         #16 vendor/rector/rector/src/Application/ApplicationFileProcessor.php(184):
         Rector\Application\FileProcessor->processFile()
         #17 vendor/rector/rector/src/Application/ApplicationFileProcessor.php(161):
         Rector\Application\ApplicationFileProcessor->processFile()
         #18 vendor/rector/rector/src/Application/ApplicationFileProcessor.php(138):
         Rector\Application\ApplicationFileProcessor->processFiles()
         #19 vendor/rector/rector/src/Console/Command/ProcessCommand.php(156):
         Rector\Application\ApplicationFileProcessor->run()
         #20 vendor/rector/rector/vendor/symfony/console/Command/Command.php(327):
         Rector\Console\Command\ProcessCommand->execute()
         #21 vendor/rector/rector/vendor/symfony/console/Application.php(960):
         RectorPrefix202410\Symfony\Component\Console\Command\Command->run()
         #22 vendor/rector/rector/vendor/symfony/console/Application.php(333):
         RectorPrefix202410\Symfony\Component\Console\Application->doRunCommand()
         #23 vendor/rector/rector/src/Console/ConsoleApplication.php(53):
         RectorPrefix202410\Symfony\Component\Console\Application->doRun()
         #24 vendor/rector/rector/vendor/symfony/console/Application.php(216):
         Rector\Console\ConsoleApplication->doRun()
         #25 vendor/rector/rector/bin/rector.php(130): RectorPrefix202410\Symfony\Component\Console\Application->run()
         #26 vendor/rector/rector/bin/rector(5): require_once('...')
         #27 vendor/bin/rector(120): include('...')
         #28 {main}". On line: 149

--- a/rules-tests/Php70/Rector/FuncCall/EregToPregMatchRector/Fixture/empty_branch.php.inc
+++ b/rules-tests/Php70/Rector/FuncCall/EregToPregMatchRector/Fixture/empty_branch.php.inc
@@ -1,0 +1,27 @@
+<?php
+
+namespace Rector\Tests\Php70\Rector\FuncCall\EregToPregMatchRector\Fixture;
+
+function emptyBranch()
+{
+	// If there was a font color or size change, change the font tag now.
+	if(@ereg("(\\\cf[0-9])||(\\\fs[0-9][0-9])", $tags)) {
+		$html .= '</font><font size="'.$size.'" color="'.$color.'">';
+	}
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\Php70\Rector\FuncCall\EregToPregMatchRector\Fixture;
+
+function emptyBranch()
+{
+	// If there was a font color or size change, change the font tag now.
+	if(@preg_match("#(\\\\cf[0-9])||(\\\\fs[0-9][0-9])#m", $tags)) {
+		$html .= '</font><font size="'.$size.'" color="'.$color.'">';
+	}
+}
+
+?>

--- a/rules/Php70/EregToPcreTransformer.php
+++ b/rules/Php70/EregToPcreTransformer.php
@@ -104,6 +104,8 @@ final class EregToPcreTransformer
         $r = [''];
         $rr = 0;
         $l = strlen($content);
+        $normalizeUnprintableChar = false;
+
         while ($i < $l) {
             // atom
             $char = $content[$i];
@@ -149,6 +151,10 @@ final class EregToPcreTransformer
                 ++$i;
                 continue;
             } elseif ($char === '|') {
+                if ($r[$rr] === '') {
+                    $normalizeUnprintableChar = true;
+                }
+
                 $r[] = '';
                 ++$rr;
                 ++$i;
@@ -185,12 +191,14 @@ final class EregToPcreTransformer
             throw new InvalidEregException('empty regular expression or branch');
         }
 
-        return [$this->normalize(implode('|', $r)), $i];
+        return [$this->normalize(implode('|', $r), $normalizeUnprintableChar), $i];
     }
 
-    private function normalize(string $content): string
+    private function normalize(string $content, bool $normalizeUnprintableChar): string
     {
-        $content = str_replace("\x0C", '\\\f', $content);
+        if ($normalizeUnprintableChar) {
+            $content = str_replace("\x0C", '\\\f', $content);
+        }
 
         return str_replace($this->pcreDelimiter, '\\' . $this->pcreDelimiter, $content);
     }

--- a/rules/Php70/EregToPcreTransformer.php
+++ b/rules/Php70/EregToPcreTransformer.php
@@ -149,10 +149,6 @@ final class EregToPcreTransformer
                 ++$i;
                 continue;
             } elseif ($char === '|') {
-                if ($r[$rr] === '') {
-                    throw new InvalidEregException('empty branch');
-                }
-
                 $r[] = '';
                 ++$rr;
                 ++$i;
@@ -194,6 +190,8 @@ final class EregToPcreTransformer
 
     private function normalize(string $content): string
     {
+        $content = str_replace("\x0C", '\\\f', $content);
+
         return str_replace($this->pcreDelimiter, '\\' . $this->pcreDelimiter, $content);
     }
 

--- a/rules/Php83/Rector/ClassMethod/AddOverrideAttributeToOverriddenMethodsRector.php
+++ b/rules/Php83/Rector/ClassMethod/AddOverrideAttributeToOverriddenMethodsRector.php
@@ -156,6 +156,7 @@ CODE_SAMPLE
                 if ($parentMethod->isPrivate()) {
                     continue;
                 }
+
                 if ($parentClassReflection->isTrait() && !$parentMethod->isAbstract()) {
                     continue;
                 }

